dietpi/func/dietpi-set_hardware

```diff
@@ -723,7 +723,14 @@ _EOF_
 		# RPi
 		if (( $G_HW_MODEL < 10 )); then
 
-			G_EXEC curl -sSfL 'https://raw.githubusercontent.com/waveshare/LCD-show/master/waveshare32b-overlay.dtb' -o /boot/overlays/waveshare32b.dtbo
+			# Download overlay to correct dir based on legacy vs modern firmware
+			local fp_overlays='/boot/firmware/overlays'
+			dpkg-query -s 'raspi-firmware' &> /dev/null || fp_overlays='/boot/overlays'
+
+			# Check if destination folder overlays exist before running curl command
+ 			[[ -d "$fp_overlays" ]] || G_EXEC mkdir -p "$fp_overlays"
```

Comment:
Actually, if this directory does not exist, there is no kernel package installed, or at least not the expected one. In that case, we better error out. So I would skip that.

The problem in your case (and for everyone with a newer RPi image) is the change boot mount point from `/boot` to `/boot/firmware`.